DSharpPlus/Net/Rest/MultipartRestRequest.cs

```diff
@@ -103,8 +105,29 @@ public HttpRequestMessage Build()
             {
                 DiscordMessageFile current = this.Files[i];
 
-                StreamContent file = new(current.Stream);
+                ArrayPoolBufferWriter<byte> writer;
 
+                try
+                {
+                    writer = new ArrayPoolBufferWriter<byte>((int)current.Stream.Length);

```

Comment:
`new APBW<byte>(checked((int)current.Stream.Length));` to preempt a nondescript runtime error; alternatively, we could also manually clamp this to the maximum size this can ever upload (100mb for bots, not sure whether bearer tokens can hit this path)

Reply:
Bearer tokens cannot currently post messages on behalf of users, no

Reply:
If we check this, we should check it some where before and throw a discriptive exception instead of fixing it